packages/plugins/opentelemetry/src/plugin.ts

```
@@ -149,268 +180,399 @@ const HeadersTextMapGetter: TextMapGetter<Headers> = {
-        tracer,
-        otelContext,
-      });
+      if (shouldTrace(options.spans?.graphql, onParamsPayload)) {
+        const { ctx, done } = startGraphQLSpan({
+          tracer,
+          params,
+          ctx: getContext(parentState),
+        });
 
-      requestContextMapping.set(request, trace.setSpan(otelContext, httpSpan));
+        forOperation.otel! = new OtelContextStack(ctx);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)